lighthouse-core/report/scripts/lighthouse-report.js

```diff
@@ -99,10 +219,37 @@ class LighthouseReport {
       });
     }
   }
+
+  /**
+   * Downloads a file (blob) using a[download].
+   * @param {Blob|File} blob The file to save.
+   */
+  _saveFile(blob) {
+    const filename = window.getFilenamePrefix({
+      url: this.json.url,
+      generatedTime: this.json.generatedTime
+    });
+
+    const ext = blob.type.match('json') ? '.json' : '.html';
+
+    const a = document.createElement('a');
+    a.download = `${filename}${ext}`;
+    a.href = URL.createObjectURL(blob);
+    a.click();
+
+    setTimeout(_ => URL.revokeObjectURL(a.href), 500); // cleanup.
+  }
 }
 
-// If in Node, allow others to require us.  By default, browser code can juse
-// use the LighthouseReport class.
-if (typeof module !== 'undefined') {
+// Allow Node code to use require(). Browser code can use the LighthouseReport
```

Comment:
Open to suggestions on this. There might be a Node-ier want to do this but I think it's simple enough.

Viewer still browserifies the code, so we keep requires for that and omit them for the browser context.